### PR TITLE
Add jfr support for monitor wait

### PR DESCRIPTION
**Pull request by tajila**

### CODE REVIEW

**Comment by tajila** [comment]
@gacholio Please review these changes

File: runtime/vm/jfr.cpp [hunk 1]
```diff
@@ -544,6 +545,40 @@ jfrVMInitialized(J9HookInterface **hook, UDATA eventNum, void *eventData, void *
...
+
+	J9JFRMonitorWaited *jfrEvent = (J9JFRMonitorWaited*)reserveBufferWithStackTrace(currentThread, currentThread, J9JFR_EVENT_TYPE_OBJECT_WAIT, sizeof(*jfrEvent));
+	if (NULL != jfrEvent) {
+		jfrEvent->time = (event->millis * 1000000) + event->nanos;
+		jfrEvent->duration = 0;
+		jfrEvent->duration = j9time_nano_time() - event->startTicks;
+		jfrEvent->monitorClass = event->monitorClass;
+		jfrEvent->monitorAddress = event->monitorAddress;
+
+		if (J9THREAD_TIMED_OUT == event->reason) {
```
Comment (gacholio) [inline runtime/vm/jfr.cpp:575]: The buffer entry is not zeroed, so you should probably just assign this unconditionally:

```
jfrEvent->timedOut = (J9THREAD_TIMED_OUT == event->reason);
```

File: runtime/vm/jfr.cpp [hunk 2]
```diff
@@ -544,6 +545,40 @@ jfrVMInitialized(J9HookInterface **hook, UDATA eventNum, void *eventData, void *
...
+	PORT_ACCESS_FROM_VMC(currentThread);
+
+#if defined(DEBUG)
+	j9tty_printf(PORTLIB, "\n!!! VM monitor waited %p\n", currentThread);
+#endif /* defined(DEBUG) */
+
+	J9JFRMonitorWaited *jfrEvent = (J9JFRMonitorWaited*)reserveBufferWithStackTrace(currentThread, currentThread, J9JFR_EVENT_TYPE_OBJECT_WAIT, sizeof(*jfrEvent));
+	if (NULL != jfrEvent) {
+		jfrEvent->time = (event->millis * 1000000) + event->nanos;
+		jfrEvent->duration = 0;
```
Comment (gacholio) [inline runtime/vm/jfr.cpp:570]: Remove this.

File: runtime/vm/threadhelp.cpp [hunk 3]
```diff
@@ -80,6 +82,12 @@ monitorWaitImpl(J9VMThread *vmThread, j9object_t object, I_64 millis, I_32 nanos
 	} else {
 		UDATA thrstate = 0;
 		J9JavaVM *javaVM = vmThread->javaVM;
+		PORT_ACCESS_FROM_JAVAVM(javaVM);
+		J9Class *monitorClass = NULL;
+		I_64 startTicks = (U_64) j9time_nano_time();
```
Comment (gacholio) [inline runtime/vm/threadhelp.cpp:87]: The cast is incorrect (and as it happens, unnecessary anyway).

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins test sanity xlinux jdk21

**Comment by gacholio** [comment]
jenkins compile win jdk8

**Pull request merged**